brewcop.py

```diff
@@ -474,27 +487,102 @@ def poll_scale(self):
 
     def tick(self):
         """
-        urwid's event loop calls this function on tick_period intervals.
-        Read the scale, then update the meter and the progress bar.
-        Switch online mode depending on weight reading.
+        urwid's event loop calls this function on tick_period_s intervals.
+        Read the scale, switch online mode depending on weight reading.
+        If online, update progress bar and offload brewing/ready heuristic
+        to the Brains class.
         """
         self.poll_scale()
         if self.scale.weight_is_valid:
             w = self.scale.weight - self.pot_tare_g
             if w < 0:
-                self.online = False
+                self.disp.offline()
             else:
                 self.disp.progress(w)
-                self.online = True
+                self.disp.online()
                 self.brains.store(w)
         self.disp.headC = self.brains.display
 
     def run(self):
         """Enter urwid's event loop.  Start ticker and handle input"""
-        self.disp.run(self.tick, self.tick_period)
-
-
-brewcop = Brewcop()
+        self.disp.run(self.tick, self.tick_period_s)
+
+
+"""
+Parse command line arguments.
+Default parameters are for Technivorm Moccamaster insulated carafe
+and scale on direct attached serial port of Raspberry Pi 2.
+"""
+
+parser = OptionParser()
```

Comment:
This is fine with me so no need to change the PR, but just as a heads up, [optparse is deprecated](https://docs.python.org/3/library/optparse.html) and argparse is the preferred alternative.

Reply:
Ooops!  I should probably fix that here.